### PR TITLE
Remove WebWorker implementation

### LINKED ISSUE #578: Remove WebWorker implementation
Remove the WebWorker implementation. The tests that use it should be moved to a Node.js execution environment (https://github.com/MetaMask/snaps-skunkworks/pull/523). Furthermore, we should stop using Electron to run tests and use JSDom for all tests.

### PR DESCRIPTION
**Pull request by FrederikBolding**
Removes the unused `WebWorker` execution environment, allowing us to remove our dependency on `electron` and allowing us to bump `jest`. Tests that were previously run with the WebWorker is now run using a Node.js execution environment. 

This PR also builds a separate Iframe testing build that is **NOT** to be used in production, but for testing purposes. This is necessary because JSDOM doesn't fully support our lockdown procedures, specifically `lockdown-more`. It also restores `fixJSDOMPostMessageEventSource.ts`.

Fixes #578

### CODE REVIEW

File: packages/controllers/jest.config.js [hunk 1]
```diff
@@ -8,39 +8,21 @@ module.exports = {
...
-        '<rootDir>/src/snaps/*',
-        '<rootDir>/src/services/*',
-      ],
-      testRegex: ['\\.test\\.(ts|js)$'],
-    },
-  ],
+  preset: 'ts-jest',
+  testRegex: ['\\.test\\.(ts|js)$'],
+  testEnvironment: 'jsdom',
+  testEnvironmentOptions: { resources: 'usable', runScripts: 'dangerously' },
```
Comment (Mrtenz) [inline packages/controllers/jest.config.js:25]: Is this safe and necessary?

File: packages/controllers/src/services/iframe/test/fixJSDOMPostMessageEventSource.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import { IframeExecutionService } from '../IframeExecutionService';
+
+// fix for jsdom does not setting event.origin or event.source: https://github.com/jsdom/jsdom/issues/2745
```
Comment (Mrtenz) [inline packages/controllers/src/services/iframe/test/fixJSDOMPostMessageEventSource.ts:3]:
```suggestion
// Fix for JSDom not setting `event.origin` or `event.source`: https://github.com/jsdom/jsdom/issues/2745
```

File: packages/execution-environments/src/common/endowments/crypto.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+const createCrypto = () => {
+  if ('crypto' in rootRealmGlobal && 'SubtleCrypto' in rootRealmGlobal) {
+    return {
+      crypto: rootRealmGlobal.crypto,
+      SubtleCrypto: rootRealmGlobal.SubtleCrypto,
+    };
+  }
+  // @todo Figure out if this is enough
+  /* eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires, node/global-require */
+  const crypto = require('crypto').webcrypto;
```
Comment (Mrtenz) [inline packages/execution-environments/src/common/endowments/crypto.ts:12]: We should add a note that this API is experimental: https://nodejs.org/api/webcrypto.html#web-crypto-api

File: packages/execution-environments/webpack.config.js [hunk 4]
```diff
@@ -15,38 +16,69 @@ module.exports = (_, argv) => {
...
+      new CopyPlugin({
+        patterns: [
           {
             from: path.resolve(
               `${path.dirname(require.resolve('ses/package.json'))}`,
               'dist',
               'lockdown.umd.min.js',
             ),
-            to: path.resolve(ENVIRONMENTS, 'node/lockdown.umd.min.js'),
+            to: path.resolve(ENVIRONMENTS, 'webworker/lockdown.umd.min.js'),
```
Comment (Mrtenz) [inline packages/execution-environments/webpack.config.js:81]: ?

File: packages/execution-environments/webpack.config.js [hunk 5]
```diff
@@ -86,5 +118,57 @@ module.exports = (_, argv) => {
...
+      alias: {
+        // Without this alias webpack tried to require ../../node_modules/stream/ which doesn't have Duplex, breaking the bundle
+        stream: 'stream-browserify',
+        'worker_threads': false
+      },
+    },
+  };
+
+  return [browserConfig, nodeConfig, unsafeConfig];
+};
```
Comment (Mrtenz) [inline packages/execution-environments/webpack.config.js:174]: Add newline.

**Review by Mrtenz — CHANGES_REQUESTED** [review]

File: packages/controllers/jest.config.js (hunk 1 above)
Comment (FrederikBolding) [reply packages/controllers/jest.config.js:25]: These `testEnvironmentOptions` are necessary to run the iframe tests, we may be able to only use these settings for that test though.

File: packages/controllers/src/services/iframe/test/fixJSDOMPostMessageEventSource.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
+import { IframeExecutionService } from '../IframeExecutionService';
+
+// Fix for JSDOM does not setting `event.origin` or `event.source`: https://github.com/jsdom/jsdom/issues/2745
```
Comment (Mrtenz) [inline packages/controllers/src/services/iframe/test/fixJSDOMPostMessageEventSource.ts:3]:
```suggestion
// Fix for JSDOM not setting `event.origin` or `event.source`: https://github.com/jsdom/jsdom/issues/2745
```

**Review by Mrtenz — CHANGES_REQUESTED** [review]

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**